### PR TITLE
Implement single origin validation through CORS 

### LINKED ISSUE #22: Restrict API to interact with the client only.
At least until it's crystal-clear that undesired actions can't be accomplished by interacting programatically with the API, its usage should be restricted to the client only.

By the way, this won't be needed while the app only has dummy data.

### PR DESCRIPTION
**Pull request by jacklagare**
Configured CORS so that only requests coming from `BASE_CLIENT_URL` are allowed. This will resolve #22.

### CODE REVIEW

**Comment by jacklagare** [comment]
Hi @AlbertoPdRF, here is the PR.

File: api/app.js [hunk 1]
```diff
@@ -26,7 +26,20 @@ connectDb(process.env.DB_URL);
...
+  origin: function (origin, callback) {
+    // If origin is the same as the value in BASE_CLIENT_URL, allow access
+    if (origin === process.env.BASE_CLIENT_URL) {
+      callback(null, true)
+    }
+    else {
+      callback(createError(403))
+    }
+  }
+}
```
Comment (AlbertoPdRF) [inline api/app.js]:
```suggestion
const corsOptions = {
  origin: (origin, callback) => {
    if (origin === process.env.BASE_CLIENT_URL) {
      callback(null, true);
    } else {
      callback(createError(403));
    }
  },
};
```
This is just a code style suggestion

**Review by AlbertoPdRF — CHANGES_REQUESTED** [review]
Nice and quick work @jacklagare! 🔝

Just a few code style issues to resolve.

**Comment by jacklagare** [comment]
@AlbertoPdRF, changes have been committed.

**Review by AlbertoPdRF — APPROVED** [review]
LGTM! Thanks for the contribution @jacklagare! 📈

**Pull request merged**